library/utils.py

```diff
@@ -1,6 +1,21 @@
 import threading
 from typing import *
-
+import logging
 
 def fire_in_thread(f, *args, **kwargs):
-    threading.Thread(target=f, args=args, kwargs=kwargs).start()
\ No newline at end of file
+    threading.Thread(target=f, args=args, kwargs=kwargs).start()
+
+def get_my_logger(name: str):
+    logger = logging.getLogger(name)
+    logger.setLevel(logging.INFO)
+
+    stream_handler = logging.StreamHandler()
+    stream_handler.setLevel(logging.INFO)
+
+    myformat = '%(asctime)s\t[%(levelname)s]\t%(filename)s:%(lineno)d\t%(message)s'
+    date_format = '%Y-%m-%d %H:%M:%S'
+    formatter = logging.Formatter(myformat, date_format)
+    stream_handler.setFormatter(formatter)
+    logger.addHandler(stream_handler)
+
+    return logger
```

Comment:
I would advise against a `get_my_logger` function like this.

The preferred way to use `logging` is to just use `logging.getLogger(__name__)` in all modules as suitable, and configure the root logger, not each logger by itself.

The log messages will then be propagated up from the named loggers into the root logger, which will print them.

Reply:
Yes, all calls of the function `get_my_logger` are like this.

```python
logger = get_my_logger(__name__)
```

So, ``name`` is ``__name__`` in ``logging.getLogger(name)``.


I think this is what you mentioned.
Please let me know if I misunderstood your intent.

Reply:
Sorry if I was unclear! I did mean that you should just use `logging.getLogger(__name__)` instead of a function that gets a logger and configures that particular logger "by hand".

Instead, it's better practice to configure the root logger, either by hand or with `logging.basicConfig()`, once per application. You can see an example of that in my PR to `stable-diffusion-webui` here: https://github.com/AUTOMATIC1111/stable-diffusion-webui/pull/14538 (though it's a bit more complex than it needs to be).

An additional issue with modifying a single logger like this PR is currently doing is that if you call `get_my_logger("something")` twice, the `"something"` logger will have two handlers, i.e. all messages logged with it will end up being printed twice.

Reply:
Thank you for your comment with an example!
I got it.
You mean that the call of logger can be like this.

```python
setup_logging() # call configuration of the root logger first
logging.getLogger(__name__) # call logger as usual way
```

Reply:
Yes, that's the idea. 😄 

`setup_logging()` would be called only once per program (so in the case of this repository, e.g. `train_network.py` and so on).